vortex-array/src/array/chunked/mod.rs

```diff
@@ -93,9 +93,10 @@ impl ChunkedArray {
     fn find_chunk_idx(&self, index: usize) -> (usize, usize) {
         assert!(index <= self.len(), "Index out of bounds of the array");
 
-        let index_chunk = search_sorted(&self.chunk_offsets(), index, SearchSortedSide::Left)
+        let index_chunk = search_sorted(&self.chunk_offsets(), index, SearchSortedSide::Right)
             .vortex_expect("Search sorted failed in find_chunk_idx")
-            .to_offset_ends_index(self.nchunks());
```

Comment:
The actual fix is here. You want to search right since with empty chunks numbers repeat and you subtract one since we start with 0

Reply:
Worth a comment IMO

Reply:
Left a comment. There was a small bug here that needed fixing. I think `to_offset_ends_index` is a made up method now that is only useful in sparse array but will clean that up in follow up